relay-crash/src/lib.rs

```diff
@@ -13,19 +14,63 @@ mod native {
     include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
 }
 
+/// A transport function used to send envelopes to Sentry.
+pub type Transport = fn(envelope: &[u8]);
+
+/// Serializes a sentry_native envelope and passes the buffer to the [`Transport`] function.
+unsafe extern "C" fn transport_proxy(
```

Comment:
We do not have access to the Sentry SDK in this crate, so instead we provide a hookable callback and inject Sentry-related logic outside.